File: src/main/java/com/geopark/web/model/entity/Researchpaper.java
package com.geopark.web.model.entity;

import com.alibaba.excel.annotation.ExcelIgnore;
import com.alibaba.excel.annotation.ExcelProperty;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;

/**
 * <p>
 * 相关科研论文
 * </p>
 *
 * @author lichee
 * @since 2019-08-29
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("researchpaper")
public class Researchpaper implements Serializable {

    @ExcelIgnore
    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    @ExcelProperty(value = "id",index = 0)
    private Integer id;

    /**
     * 地质公园编号
     */
    @ExcelProperty(value = "公园编号",index = 1)
    private Integer parkId;

    /**
     * 成果编号
     */
    @ExcelProperty(value = "成果编号",index = 2)
    private String achievementnumber;

    /**
     * 论文编号
     */
    @ExcelProperty(value = "论文编号",index = 3)
    private String papernumber;

    /**
     * 作者
     */
    @ExcelProperty(value = "作者",index = 4)
    private String author;

    /**
     * 年份
     */
    @ExcelProperty(value = "年份",index = 5)
    private String year;

    /**
     * 名称
     */
    @ExcelProperty(value = "名称",index = 6)
    private String heading;

    /**
     * 期刊
     */
    @ExcelProperty(value = "期刊",index = 7)
    private String periodical;

    /**
     * 卷期
     */
    @ExcelProperty(value = "卷期",index = 8)
    private String volume;

    /**
     * 页码
     */
    @ExcelProperty(value = "页码",index = 9)
    private String pagenumber;

    /**
     * 摘要
     */
    @ExcelProperty(value = "摘要",index = 10)
    private String summary;

    /**
     * 收集情况
     */
    @ExcelProperty(value = "收集情况",index = 11)
    private String collect;

    /**
     * 备注
     */
    @ExcelProperty(value = "备注",index = 12)
    private String note;

    /**
     * 存储地址
     */
    @ExcelProperty(value = "存储地址",index = 13)
    private String url;

}

File: src/main/resources/static/js/science/researchpaper.js
//公园概况

layui.use(['config', 'lichee', 'jquery', 'layer', 'table', 'form', 'laydate', 'fileUpload'], function () {

    var config = layui.config;
    var lichee = layui.lichee;
    var $ = layui.jquery;
    var layer = layui.layer;
    var table = layui.table;
    var form = layui.form;
    var laydate = layui.laydate;
    var fileUpload = layui.fileUpload;

    var tableInfo = table.render({
        elem: '#table',
        toolbar: '#toolbar',
        defaultToolbar: [],
        //width:'auto',
        url: '/researchpaper/page',
        title: '科研论文',
        page: true,
        headers: {Authorization: config.getToken()},
        request: config.request,
        parseData: config.parseData,
        response: config.response,
        cols: [[
            //{type: 'checkbox', fixed: 'left'},
            // {field: 'id', title: 'ID'},
            {field: 'parkId', align: 'center', sort: true, title: '地质公园ID'},
            {field: 'achievementnumber', align: 'center', sort: true, title: '成果编号'},
            {field: 'papernumber', align: 'center', sort: true, title: '论文编号'},
            {field: 'author', align: 'center', sort: true, title: '作者'},
            {field: 'year', align: 'center', sort: true, title: '年份'},
            {field: 'heading', align: 'center', sort: true, title: '名称'},
            {field: 'periodical', align: 'center', sort: true, title: '期刊'},
            {field: 'volume', align: 'center', sort: true, title: '卷期'},
            {field: 'pagenumber', align: 'center', sort: true, title: '页码'},
            //{field: 'summary', align: 'center', sort: true, title: '摘要'},
            //{field: 'collect', align: 'center', sort: true, title: '收集情况'},
            //{field: 'note', align: 'center', sort: true, title: '备注'},
            {fixed: 'right',align : 'center', title: '操作', toolbar: '#bar', width: 180}
        ]]
    });

    table.on('toolbar(tableFilter)', function (obj) {
        var checkStatus = table.checkStatus(obj.config.id);
        switch (obj.event) {
            case 'add':
                showEditModel();
                break;
            case 'getCheckLength':
                var data = checkStatus.data;
                layer.msg('选中了：' + data.length + ' 个');
                break;
            case 'isAll':
                layer.msg(checkStatus.isAll ? '全选' : '未全选');
                break;
        };
    });

    table.on('tool(tableFilter)', function (obj) {
        var data = obj.data;
        if (obj.event === 'del') {
            layer.confirm('确定要删除吗？', function (index) {
                layer.load(2);
                lichee.delete('/researchpaper/' + obj.data.id, {}, function () {
                    layer.closeAll('loading');
                    layer.msg('删除成功', {icon: 1});
                    obj.del();
                });
            });
        } else if (obj.event === 'edit') {
            showEditModel(data);
        }
    });

    var showEditModel = function (data) {
        layer.open({
            title: data ? '详情/修改' : '添加',
            type: 1,
            area: ['800px', '600px'],
            content: $('#form-model').html(),
            success: function () {
                $('#form')[0].reset();
                lichee.get('/park/list', {async: false}, function (data) {
                    $('#parkId').vm({parks: data.result});
                    form.render('select');
                });
                fileUpload.init("uploadBtn", "url", "researchpaper");
                if (data) {
                    form.val('formFilter', data);
                    fileUpload.initFileList(data.url);
                }
                $('#form .close').click(function () {
                    layer.closeAll('page');
                });
            }
        });
    };

    var form = layui.form;
    form.on('submit(formSubmit)', function (data) {
        layer.load(2);
        if (data.field.id) {
            lichee.put('/researchpaper/' + data.field.id, {data: data.field}, function (res) {
                callFunction(res);
            });
        } else {
            lichee.post('/researchpaper', {data: data.field}, function (res) {
                callFunction(res);
            });
        }
        return false;
    });

    form.verify({
        commonLength: [/^[\S]{1,100}$/, '字符长度最大100'],
        descLength: [/^[\S]{0,180}$/, '字符长度最大180']
    });

    var callFunction = function (res) {
        layer.closeAll();
        if (res.status == 200) {
            layer.msg("修改成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        } else if(res.status == 201)  {
            layer.msg("添加成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        }
    };

    $('#searchButton').click(function () {
        tableInfo.reload({where: lichee.getSearchForm()});
    });

    laydate.render({
        elem: '#startTime',
        type: 'datetime',
        value: '',
        trigger: 'click'
    });

    $(".laydate").each(function (i,e) {
        laydate.render({
            elem: this,
            type: 'datetime',
            value: '',
            trigger: 'click'
        });
    });

});